src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -388,7 +388,11 @@ export function loadOperation(
         if (op.verb !== "delete") {
             const formattedType = getFormattedType(
                 program,
-                metadata.logicalResult
+                // TODO: we should check `logicalPath` or other ways to determine body type,
+                // after https://github.com/Azure/typespec-azure/issues/3725 is fixed
+                op.verb === "post"
```

Comment:
How about `op.verb !== "post"` operation? Can a `put` Lro operation has `envelopeResult`?

Reply:
> How about `op.verb !== "post"` operation? Can a `put` Lro operation has `envelopeResult`?

From the spec, `logicalPath` is for `post` (e.g. RPC), at least for now. That's why I'm using `=== "post"`.

Reply:
The better way is to check `logicalPath` regardless of HTTP verb. But there is an issue here.